discord/commands/context.py

```diff
@@ -83,85 +78,74 @@ class ApplicationContext(discord.abc.Messageable):
-        This is useful if you want to just call the callback that a
-        :class:`.ApplicationCommand` holds internally.
+        This is similar to :meth:`~.BaseContext.invoke` except that it bypasses
+        checks, cooldowns, and error handlers.
 
         .. note::
 
-            This does not handle converters, checks, cooldowns, pre-invoke,
-            or after-invoke hooks in any matter. It calls the internal callback
-            directly as-if it was a regular function.
-            You must take care in passing the proper arguments when
-            using this function.
+            If you want to bypass :exc:`.UserInputError` derived exceptions,
+            it is recommended to use the regular :meth:`~.Context.invoke`
+            as it will work more naturally. After all, this will end up
+            using the old arguments the user has used and will thus just
+            fail again.
 
         Parameters
-        -----------
-        command: :class:`.ApplicationCommand`
-            The command that is going to be called.
-        \*args
-            The arguments to use.
-        \*\*kwargs
-            The keyword arguments to use.
+        ------------
+        call_hooks: :class:`bool`
+            Whether to call the before and after invoke hooks.
+        restart: :class:`bool`
+            Whether to start the call chain from the very beginning
+            or where we left off (i.e. the command that caused the error).
+            The default is to start where we left off.
 
         Raises
         -------
-        TypeError
-            The command argument to invoke is missing.
+        ValueError
+            The context to reinvoke is not valid.
         """
-        return await command(self, *args, **kwargs)
-
-    @cached_property
-    def channel(self) -> Optional[InteractionChannel]:
-        """Union[:class:`abc.GuildChannel`, :class:`PartialMessageable`, :class:`Thread`]:
-        Returns the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel`."""
-        return self.interaction.channel
+        cmd = self.command
+        if cmd is None:
+            raise ValueError("This context is not valid.")
 
-    @cached_property
-    def channel_id(self) -> Optional[int]:
-        """:class:`int`: Returns the ID of the channel associated with this context's command.
-        Shorthand for :attr:`.Interaction.channel_id`.
-        """
-        return self.interaction.channel_id
+        if restart:
+            to_call = cmd.root_parent or cmd
+        else:
+            to_call = cmd
 
-    @cached_property
-    def guild(self) -> Optional[Guild]:
-        """Optional[:class:`.Guild`]: Returns the guild associated with this context's command.
-        Shorthand for :attr:`.Interaction.guild`.
-        """
-        return self.interaction.guild
+        try:
+            await to_call.reinvoke(self, call_hooks=call_hooks)
+        finally:
+            self.command = cmd
```

Comment:
I'm a little confused on why `self.command` is set to `cmd` which was set to `self.command`?

Reply:
The attribute may get edited here:

https://github.com/Pycord-Development/pycord/blob/57e257dec4124c42a61f08a856ec6a5f8ecc7c57/discord/ext/commands/core.py#L953-L954

https://github.com/Pycord-Development/pycord/blob/57e257dec4124c42a61f08a856ec6a5f8ecc7c57/discord/ext/commands/core.py#L1558-L1562